Tasks/CopyFilesV2/Tests/L0preservesTimestampIfSpecified.ts

```diff
@@ -0,0 +1,54 @@
+import fs = require('fs');
+import mockanswer = require('vsts-task-lib/mock-answer');
+import mockrun = require('vsts-task-lib/mock-run');
+import path = require('path');
+
+let taskPath = path.join(__dirname, '..', 'copyfiles.js');
+let runner: mockrun.TaskMockRunner = new mockrun.TaskMockRunner(taskPath);
+runner.setInput('Contents', '**');
+runner.setInput('SourceFolder', path.normalize('/srcDir'));
+runner.setInput('TargetFolder', path.normalize('/destDir'));
+runner.setInput('CleanTargetFolder', 'false');
+runner.setInput('Overwrite', 'false');
+runner.setInput('preserveTimestamp', 'true');
+let answers = <mockanswer.TaskLibAnswers>{
+    checkPath: {},
+    find: {},
+};
+answers.checkPath[path.normalize('/srcDir')] = true;
+answers.find[path.normalize('/srcDir')] = [
+    path.normalize('/srcDir'),
+    path.normalize('/srcDir/someOtherDir'),
+    path.normalize('/srcDir/someOtherDir/file1.file'),
+    path.normalize('/srcDir/someOtherDir/file2.file'),
+    path.normalize('/srcDir/someOtherDir2'),
+    path.normalize('/srcDir/someOtherDir2/file1.file'),
+    path.normalize('/srcDir/someOtherDir2/file2.file'),
+    path.normalize('/srcDir/someOtherDir2/file3.file'),
+    path.normalize('/srcDir/someOtherDir3'),
+];
+runner.setAnswers(answers);
+runner.registerMockExport('stats', (itemPath: string) => {
+    console.log('##vso[task.debug]stats ' + itemPath);
+    switch (itemPath) {
+        case path.normalize('/srcDir/someOtherDir'):
+        case path.normalize('/srcDir/someOtherDir2'):
+        case path.normalize('/srcDir/someOtherDir3'):
+            return { isDirectory: () => true };
+        case path.normalize('/srcDir/someOtherDir/file1.file'):
+        case path.normalize('/srcDir/someOtherDir/file2.file'):
+        case path.normalize('/srcDir/someOtherDir2/file1.file'):
+        case path.normalize('/srcDir/someOtherDir2/file2.file'):
+        case path.normalize('/srcDir/someOtherDir2/file3.file'):
+            return { isDirectory: () => false };
+        default:
+            throw { code: 'ENOENT' };
+    }
+});
+
+fs.utimes = function (targetPath, atime, mtime, err) {
```

Comment:
Can we do something like `console.log('Calling fs.utimes on', targetPath)`. Then in the actual L0 test we can check the stdout to make sure that the correct files are getting updated - see [here](https://github.com/Microsoft/azure-pipelines-tasks/blob/c123e38cff7aa7f4b05de4fe8beda3e46c0a48b3/Tasks/UseNodeV1/Tests/L0.ts#L75) for an example of checking stdout

Reply:
ah, the asserts are over there, right. I'm starting to get the hang of this. I added that log and verified it is showing during the assert.

Reply:
Looks good! Added one nit, once that's in I think we can merge!